spec/lib/tasks/provider_transfer/ptr_migrations_spec.rb

```diff
@@ -0,0 +1,179 @@
+        expect(Rails.logger).to receive(:info).with("Provider (#{provider.email}) successfully transferred to firm #{new_firm.name}")
+
+        task.execute(mock: "false", providerid: provider.id, firmid: new_firm.id)
+
+        expect(provider.reload.firm).to eq(new_firm)
+        expect(provider.reload.selected_office_id).to be_nil
+      end
+    end
+
+    context "when Provider does not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("PROVIDER or FIRM does not exist")
+
+        task.execute(mock: "false", providerid: nil, firmid: new_firm.id)
+      end
+    end
+
+    context "when Firm does not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("PROVIDER or FIRM does not exist")
+
+        task.execute(mock: "false", providerid: provider.id, firmid: nil)
+      end
+    end
+
+    context "when both Provider and Firm do not exist" do
+      it "logs an error message" do
+        expect(Rails.logger).to receive(:info).with("PROVIDER or FIRM does not exist")
+
+        task.execute(mock: "false", providerid: nil, firmid: nil)
+      end
+    end
+
+    context "when mock is true" do
+      it "logs mock transfer messages without updating the records" do
+        expect(Rails.logger).to receive(:info).with("Provider (#{provider.email}) current firm #{initial_firm.name}")
+        expect(Rails.logger).to receive(:info).with("Provider (#{provider.email}) successfully transferred to firm #{initial_firm.name}")
+
+        task.execute(mock: "true", providerid: provider.id, firmid: new_firm.id)
+
+        expect(provider.reload.firm).to eq(initial_firm)
+      end
+    end
+  end
+
+  describe "delete_firm_office" do
+    subject(:task) do
+      Rake::Task["ptr_migrations:delete_firm_office"]
+    end
+
+    before do
+      firm.offices << office
+    end
+
+    context "when Firm exists and has no associated LegalAidApplications" do
+      it "deletes the Office and the Firm" do
+        expect(Rails.logger).to receive(:info).with("Firm (#{firm.name}), Office (#{office.code}) deleted successfully")
+        expect(Rails.logger).to receive(:info).with("Firm (#{firm.name}), deleted successfully")
+
+        task.execute(mock: "false", firmid: firm.id)
+
+        expect(Office).not_to exist(office.id)
+        expect(Firm).not_to exist(firm.id)
+      end
+    end
+
+    context "when Firm exists and Office has associated LegalAidApplications" do
+      before do
+        create(:legal_aid_application, office: office)
+      end
+
+      it "skips deleting the Office and logs a message" do
+        expect(Rails.logger).to receive(:info).with("Firm (#{firm.name}), Office (#{office.code}) has 1 legal aid application associated with it. Skipping!")
+        expect(Rails.logger).to receive(:info).with("Firm (#{firm.name}) has 1 office associated with it. Skipping!")
```

Comment:
Is this supposed to work when there are multiple offices? Could there a be a test for this scenario?

Reply:
@agoldstone93 spec updated